engine/api/src/main/java/io/deephaven/engine/table/updateBySpec/CumProdSpec.java

```diff
@@ -1,20 +1,18 @@
 package io.deephaven.engine.table.updateBySpec;
 
+import io.deephaven.annotations.SimpleStyle;
 import io.deephaven.util.type.TypeUtils;
+import org.immutables.value.Value.Immutable;
 import org.jetbrains.annotations.NotNull;
 
 /**
  * A {@link UpdateBySpec} for performing a Cumulative Product of the specified columns.
  */
+@Immutable
+@SimpleStyle
 public class CumProdSpec implements UpdateBySpec {
```

Comment:
I'm a little surprised this worked - but CumProdSpec should be abstract as well.